base/compiler/utilities.jl

```diff
@@ -118,8 +118,24 @@ function retrieve_code_info(linfo::MethodInstance)
     end
 end
 
+# Get at the nonfunction_mt, which happens to be the mt of SimpleVector
+const nonfunction_mt = typename(SimpleVector).mt
+
+function get_compileable_sig(method::Method, @nospecialize(atypes), sparams::SimpleVector)
+    isa(atypes, DataType) || return Nothing
+    mt = ccall(:jl_method_table_for, Any, (Any,), atypes)
+    mt === nothing && return nothing
+    return ccall(:jl_normalize_to_compilable_sig, Any, (Any, Any, Any, Any),
+        mt, atypes, sparams, method)::Union{Nothing, Type{<:Tuple}}
```

Comment:
This is a pretty expensive looking type-assert, since it also conveys no useful information

Reply:
Fair enough. I'll drop it.